### PR TITLE
Bug fixes in Persson/Peraire shock sensor.

### PR DESCRIPTION
**Pull request by tarikdzanic**

### CODE REVIEW

**Comment by tarikdzanic** [comment]
Updated syntax.

File: pyfr/solvers/baseadvecdiff/elements.py [hunk 1]
```diff
@@ -111,13 +111,19 @@ def gradcoru_qpts():
             shockvar = self.convarmap[self.ndims].index(self.shockvar)
 
             # Obtain the degrees of the polynomial modes in the basis
-            ubdegs = [sum(dd) for dd in self.basis.ubasis.degrees]
+            ubasis = self.basis.ubasis
+            ubdegs_p = ubasis.degrees
+            # Obtain the degrees of the polynomial modes in the basis projected to one order lower
+            basis_type = type(ubasis)
+            ubdegs_pm1 = basis_type(max(0, ubasis.order-1), ubasis.pts).degrees
```
Comment (FreddieWitherden) [inline pyfr/solvers/baseadvecdiff/elements.py:118]: get_polybasis(ubasis.name, max(0, ubasis.order - 1)).degrees

File: pyfr/solvers/baseadvecdiff/kernels/shocksensor.mako [hunk 2]
```diff
@@ -2,22 +2,22 @@
...
+% for ivdm, bmode in zip(invvdm, ind_modes):
     tmp = ${' + '.join('{jx}*u[{j}][{svar}]'.format(j=j, jx=jx, svar=svar)
-                       for j, jx in enumerate(invvdm[i]) if jx != 0)};
+                       for j, jx in enumerate(ivdm) if jx != 0)};
 
     totEn += tmp*tmp;
-% if deg >= order:
-    pnEn += tmp*tmp;
-% endif
+    % if bmode:
```
Comment (FreddieWitherden) [inline pyfr/solvers/baseadvecdiff/kernels/shocksensor.mako:18]: Indentation.

File: pyfr/solvers/baseadvecdiff/elements.py [hunk 3]
```diff
@@ -2,6 +2,7 @@
 
 from pyfr.backends.base.kernels import ComputeMetaKernel
 from pyfr.solvers.baseadvec import BaseAdvectionElements
+from pyfr.polys import get_polybasis
```
Comment (FreddieWitherden) [inline pyfr/solvers/baseadvecdiff/elements.py:5]: Move up so alphabetically ordered.

**Pull request merged**